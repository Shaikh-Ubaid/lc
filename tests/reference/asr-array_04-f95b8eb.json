{
    "basename": "asr-array_04-f95b8eb",
    "cmd": "lc --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/array_04.cpp",
    "infile_hash": "86bab3bcf300c07e9861e92ce21215829ae2893db1d378776b8ab332",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_04-f95b8eb.stdout",
    "stdout_hash": "b63d1dcba8f9628e897f64e5ed5f57df9da1c398131da94d0a722bee",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}