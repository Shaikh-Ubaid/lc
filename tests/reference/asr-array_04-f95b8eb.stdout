(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr1:
                                (Variable
                                    2
                                    arr1
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr2:
                                (Variable
                                    2
                                    arr2
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            res:
                                (Variable
                                    2
                                    res
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            res_:
                                (Variable
                                    2
                                    res_
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Real 8)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 3 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(Allocate
                        [((Var 2 arr1)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))
                        ((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (Assignment
                        (Var 2 arr1)
                        (ArrayReshape
                            (ArrayConstant
                                [(RealConstant
                                    1.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    3.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )]
                                (Array
                                    (Real 8)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 9 (Integer 4)))]
                                    FixedSizeArray
                                )
                                RowMajor
                            )
                            (ArrayPhysicalCast
                                (ArrayConstructor
                                    [(ArraySize
                                        (Var 2 arr1)
                                        (IntegerConstant 1 (Integer 4))
                                        (Integer 4)
                                        ()
                                    )
                                    (ArraySize
                                        (Var 2 arr1)
                                        (IntegerConstant 2 (Integer 4))
                                        (Integer 4)
                                        ()
                                    )]
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 2 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    RowMajor
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 2 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )
                            (Array
                                (Real 8)
                                [(()
                                ())
                                (()
                                ())]
                                FixedSizeArray
                            )
                            ()
                        )
                        ()
                    )
                    (Allocate
                        [((Var 2 arr2)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (Assignment
                        (Var 2 arr2)
                        (ArrayConstant
                            [(RealConstant
                                5.000000
                                (Real 8)
                            )
                            (RealConstant
                                6.000000
                                (Real 8)
                            )
                            (RealConstant
                                7.000000
                                (Real 8)
                            )]
                            (Array
                                (Real 8)
                                [((IntegerConstant 0 (Integer 4))
                                (IntegerConstant 3 (Integer 4)))]
                                FixedSizeArray
                            )
                            RowMajor
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 res_)
                        (ArrayConstant
                            [(RealConstant
                                7.000000
                                (Real 8)
                            )
                            (RealConstant
                                11.000000
                                (Real 8)
                            )
                            (RealConstant
                                14.000000
                                (Real 8)
                            )]
                            (Array
                                (Real 8)
                                [((IntegerConstant 0 (Integer 4))
                                (IntegerConstant 3 (Integer 4)))]
                                FixedSizeArray
                            )
                            RowMajor
                        )
                        ()
                    )
                    (Allocate
                        [((Var 2 res)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (Assignment
                        (Var 2 res)
                        (RealBinOp
                            (ArraySection
                                (Var 2 arr1)
                                [(()
                                (IntegerConstant 1 (Integer 4))
                                ())
                                ((ArrayBound
                                    (Var 2 arr1)
                                    (IntegerConstant 2 (Integer 4))
                                    (Integer 4)
                                    LBound
                                    ()
                                )
                                (ArrayBound
                                    (Var 2 arr1)
                                    (IntegerConstant 2 (Integer 4))
                                    (Integer 4)
                                    UBound
                                    ()
                                )
                                (IntegerConstant 1 (Integer 4)))]
                                (Array
                                    (Real 8)
                                    [(()
                                    ())]
                                    DescriptorArray
                                )
                                ()
                            )
                            Add
                            (Var 2 arr2)
                            (Array
                                (Real 8)
                                [(()
                                ())]
                                DescriptorArray
                            )
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 2 arr1)
                        (StringConstant
                            "\n"
                            (Character 1 1 ())
                        )]
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                    )
                    (Print
                        [(Var 2 arr2)
                        (StringConstant
                            "\n"
                            (Character 1 1 ())
                        )]
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                    )
                    (Print
                        [(Var 2 res)
                        (StringConstant
                            "\n"
                            (Character 1 1 ())
                        )]
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                        (StringConstant
                            ""
                            (Character 1 1 ())
                        )
                    )
                    (If
                        (IntrinsicArrayFunction
                            Any
                            [(RealCompare
                                (IntrinsicElementalFunction
                                    Abs
                                    [(RealBinOp
                                        (Var 2 res)
                                        Sub
                                        (Var 2 res_)
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        ()
                                    )]
                                    0
                                    (Array
                                        (Real 8)
                                        [(()
                                        ())]
                                        DescriptorArray
                                    )
                                    ()
                                )
                                Gt
                                (ArrayBroadcast
                                    (RealConstant
                                        0.000000
                                        (Real 8)
                                    )
                                    (IntrinsicArrayFunction
                                        Shape
                                        [(IntrinsicElementalFunction
                                            Abs
                                            [(RealBinOp
                                                (Var 2 res)
                                                Sub
                                                (Var 2 res_)
                                                (Array
                                                    (Real 8)
                                                    [(()
                                                    ())]
                                                    DescriptorArray
                                                )
                                                ()
                                            )]
                                            0
                                            (Array
                                                (Real 8)
                                                [(()
                                                ())]
                                                DescriptorArray
                                            )
                                            ()
                                        )]
                                        0
                                        (Array
                                            (Integer 4)
                                            [((IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 1 (Integer 4)))]
                                            FixedSizeArray
                                        )
                                        ()
                                    )
                                    (Array
                                        (Real 8)
                                        [(()
                                        ())]
                                        DescriptorArray
                                    )
                                    ()
                                )
                                (Array
                                    (Logical 4)
                                    [(()
                                    ())]
                                    DescriptorArray
                                )
                                ()
                            )]
                            0
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            (IntegerConstant 2 (Integer 4))
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            exit_code:
                                (Variable
                                    3
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(Assignment
                        (Var 3 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
